lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -259,6 +260,39 @@ defmodule ExUnit.Runner do
     end
   end
 
+  defp maybe_create_tmp_dir(test, context, %{tmp_dir: true}) do
+    create_tmp_dir!(test, "", context)
+  end
+
+  defp maybe_create_tmp_dir(test, context, %{tmp_dir: path}) when is_binary(path) do
+    create_tmp_dir!(test, path, context)
+  end
+
+  defp maybe_create_tmp_dir(_, _, %{tmp_dir: other}) do
+    raise ArgumentError, "expected :tmp_dir to be atom true or a string, got: #{inspect(other)}"
+  end
+
+  defp maybe_create_tmp_dir(_, context, _) do
+    context
+  end
+
+  defp create_tmp_dir!(test, extra_path, context) do
+    module = escape_path(inspect(test.module))
+    name = escape_path(to_string(test.name))
+    path = Path.join(["tmp", module, name, extra_path])
+    File.rm_rf!(path)
+    File.mkdir_p!(path)
+    Map.put(context, :tmp_dir, path)
+  end
+
+  defp escape_path(path) do
+    case :os.type() do
+      {:win32, _} -> String.replace(path, ~R'[~#%&*{}\\:<>?/+|"]', "_")
+      _ -> path
+    end
+    |> String.replace("/", "_")
```

Comment:
I would always run the escape characters. I also think the escape characters can be written like this:

```
@escape Enum.map('[~#%&*{}\\:<>?/+|"]', <<&1::utf8>>)
```

Reply:
Then you can do `String.replace(test, @escape, "_")`.

Reply:
Good call. I decided to still escape things individually, that is we have: `tmp/MyTest/test my test`, as opposed to: `tmp/MyTest_test my test` (`/` would have been escaped)